docs-source/content/faq/oci-lb.md

```diff
@@ -0,0 +1,100 @@
+the default shape (100Mbps) will be used.
+
+```
+apiVersion: v1
+kind: Service
+metadata:
+  name: bobs-bookstore-oci-lb-service
+  namespace: bob
+  annotations:
+    service.beta.kubernetes.io/oci-load-balancer-shape: 400Mbps
+spec:
+  ports:
+  - name: http
+    port: 31111
+    protocol: TCP
+    targetPort: 31111
+  selector:
+    weblogic.clusterName: cluster-1
+    weblogic.domainUID: bobs-bookstore
+  sessionAffinity: None
+  type: LoadBalancer
+```
+
+When you apply this YAML file to your cluster, you will see the new service is created
+but initially the external IP is shown as `<pending>`.  
+
+```
+$ kubectl -n bob get svc
+NAME                                   TYPE           CLUSTER-IP      EXTERNAL-IP   PORT(S)                       AGE
+bobs-bookstore-admin-server            ClusterIP      None            <none>        8888/TCP,7001/TCP,30101/TCP   9d
+bobs-bookstore-admin-server-external   NodePort       10.96.224.13    <none>        7001:32401/TCP                9d
+bobs-bookstore-cluster-cluster-1       ClusterIP      10.96.86.113    <none>        8888/TCP,8001/TCP,31111/TCP   9d
+bobs-bookstore-managed-server1         ClusterIP      None            <none>        8888/TCP,8001/TCP,31111/TCP   9d
+bobs-bookstore-managed-server2         ClusterIP      None            <none>        8888/TCP,8001/TCP,31111/TCP   9d
+bobs-bookstore-oci-lb-service          LoadBalancer   10.96.121.216   <pending>     31111:31671/TCP               9s
+```
+
+After a short time (typically less than a minute) the OCI Load Balancer will be provisioned and the
+external IP address will be displayed:
+
+```
+$ kubectl -n bob get svc
+NAME                                   TYPE           CLUSTER-IP      EXTERNAL-IP       PORT(S)                       AGE
+bobs-bookstore-admin-server            ClusterIP      None            <none>            8888/TCP,7001/TCP,30101/TCP   9d
+bobs-bookstore-admin-server-external   NodePort       10.96.224.13    <none>            7001:32401/TCP                9d
+bobs-bookstore-cluster-cluster-1       ClusterIP      10.96.86.113    <none>            8888/TCP,8001/TCP,31111/TCP   9d
+bobs-bookstore-managed-server1         ClusterIP      None            <none>            8888/TCP,8001/TCP,31111/TCP   9d
+bobs-bookstore-managed-server2         ClusterIP      None            <none>            8888/TCP,8001/TCP,31111/TCP   9d
+bobs-bookstore-oci-lb-service          LoadBalancer   10.96.121.216   132.145.235.215   31111:31671/TCP               55s
+```
+
+You can now use the external IP address and port to access your pods.  There are several
+options that can be used to configure more advanced load balancing behavior.  Please 
+refer to the OCI documentation](https://docs.cloud.oracle.com/iaas/Content/ContEng/Tasks/contengcreatingloadbalancer.htm)
+for more information, including how to configure SSL support, supporting internal and external subnets, and so on.
+
```

Comment:
Does the user have to create the OCI LoadBalance first (e.g. from the OCI console) or does OKE create it.

Reply:
OKE automatically creates it  - see lines 8 and 82. Do you think I need to make that clearer?

Reply:
No, they don't.  OCI will create the load balancer.  I'd be interested in trying if you could create the load balancer in the OCI console and then "see it" or reference it here.

One thing that might be useful...  I'm pretty sure that after OCI provisions an IP address the first time, then you could use that IP address in your yaml to make sure that if the load balancer was deleted then it would get recreated with the same IP.

Reply:
Yes I can see where you do indicate that, I just missed it.  I am wondering about the details, like what compartment, what subnet did it use etc. That is why I sort of assumed the user had to do it first.  I guess it must create a new public subnet (s) or create a regional subnet and setup the sec rules so the LB can talk to the pods.

Reply:
No it just puts it on the LB subnets (2) for the OKE cluster

Reply:
All that other stuff is pre-created when you create the cluster itself

Reply:
Is there anyway to use a single OCI LB across 2 domains or is it 1 LB per domain.

Reply:
i believe you can set up host based routing..

Reply:
actually not sure about that

Reply:
i think it might be a 1:1 relationship between the service and the LB instance

Reply:
nice article that is a pretty cool feature

Reply:
You can do the 1:many relationship using an ingress controller, but not for a straight, LoadBalancer service.